src/client/main.tsx

```diff
@@ -1,11 +1,31 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { css, Global } from '@emotion/core';
 import { fontFaces } from '@/client/lib/fonts';
 import { ClientStateProvider } from '@/client/components/ClientState';
 import { ClientState } from '@/shared/model/ClientState';
 import { GatewayRoutes } from './routes';
+import { tests } from '@/shared/model/experiments/abTests';
+import { useAB } from '@guardian/ab-react';
+// Uncomment below line for AB Test Demo
+// import { ABTestDemo } from './components/ABTestDemo';
 
 export const Main = (props: ClientState) => {
+  // initalise the AB Test Framework:
+  // load the AB Hook
+  const ABTestAPI = useAB();
+
+  // use effect to initialise and register events if needed
+  useEffect(() => {
```

Comment:
I only about 50% sure here but I'm wondering if this `useEffect` is needed?

Reply:
I mainly followed the guide from https://github.com/guardian/ab-testing#initialising-in-a-react-project

From what I can tell the `useEffect` prevents the tracking/completion/impression events from firing more than once, as we only need them on the initial load.